CONTRIBUTING.md

```diff
@@ -0,0 +1,246 @@
+# Contributing
+
+The following documentation is only for the maintainers of this repository.
+
+- [Monorepo setup](#monorepo-setup)
+- [Project overview](#project-overview)
+- [Installation](#installation)
+- [Develop the CLI packages](#Develop-the-CLI-packages)
+- [Release the packages](#release-the-packages)
+- [Available commands](#commands)
+- [CI](#ci)
+- [Add a new package to the monorepo](#add-a-new-package-to-the-monorepo)
+
+## Monorepo setup
+
+This repository is managed as a monorepo with [PNPM workspace](https://pnpm.io/workspaces) to handle the installation of the npm dependencies and manage the packages interdependencies.
+
+It's important to note that PNPM workspace doesn't hoist the npm dependencies at the root of the workspace as most package manager does. Instead, it uses an advanced [symlinked node_modules structure](https://pnpm.io/symlinked-node-modules-structure). This means that you'll find a `node_modules` directory inside the packages folders as well as at the root of the repository.
+
+The main difference to account for is that the `devDependencies` must now be installed locally in every package `package.json` file rather than in the root `package.json` file.
+
+## Project overview
+
+This project is split into two major sections, [packages/](packages/) and [sample/](sample/).
+
+### Packages
+
+Under [packages/](packages/) are the actual packages composing the federated application shell.
+
+[@squide/core](packages/core/) is a package including the core functionalities of the shell, like the runtime and the messaging infrastructure. The shell architecture is very similar to an [hexagonal architecture](https://en.wikipedia.org/wiki/Hexagonal_architecture_(software)), whereas the ports are mostly what consistitute the core package.
+
+[@squide/react-router](packages/react-router/) is a [React Router](https://reactrouter.com/en/main) implementation of the shell routing capabilities. This implementation is offered as a standalone package because the shell could eventually support alternative routing libraries like [TanStack router](https://tanstack.com/router/v1).
+
+[@squide/webpack-module-federation](packages/webpack-module-federation/) is module federation implementation for [Webpack](https://webpack.js.org/concepts/module-federation/). This implementation is offered as a standalone package because not all application configuration will require module federation and the shell could eventually support alternative module federation application like [Rspack](https://www.rspack.dev/).
```

Comment:
```suggestion
[@squide/webpack-module-federation](packages/webpack-module-federation/) is module federation implementation for [Webpack](https://webpack.js.org/concepts/module-federation/). This implementation is offered as a standalone package because not all application configurations will require module federation and the shell could eventually support alternative module federation application like [Rspack](https://www.rspack.dev/).
```